activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -486,6 +470,36 @@ def serialize_entries(entries, **options)
           end
         end
 
+        def change_counter(key, amount, options)
+          redis.with do |c|
+            c = c.node_for(key) if c.is_a?(Redis::Distributed)
+
+            if options[:expires_in] && supports_expire_nx?
+              c.pipelined do |p|
+                p.incrby(key, amount)
+                p.call(:expire, key, options[:expires_in].to_i, "NX")
+              end.first
+            else
+              count = c.incrby(key, amount)
+              if options[:expires_in] && c.ttl(key) < 0
+                c.expire(key, options[:expires_in].to_i)
+              end
+              count
+            end
+          end
+        end
+
+        def supports_expire_nx?
+          redis_version >= Gem::Version.new("7.0.0")
```

Comment:
I think we should memoize the boolean as allocating `Gem::Version` and doing the check every time is a bit wasteful.